Makefile

```diff
@@ -330,6 +330,39 @@ cyghts-$(LIBHTS_SOVERSION).dll libhts.dll.a: $(LIBHTS_OBJS)
 hts-$(LIBHTS_SOVERSION).dll hts.dll.a: $(LIBHTS_OBJS)
 	$(CC) -shared -Wl,--out-implib=hts.dll.a -Wl,--enable-auto-import -Wl,--exclude-all-symbols $(LDFLAGS) -o $@ -Wl,--whole-archive $(LIBHTS_OBJS) -Wl,--no-whole-archive $(LIBS) -lpthread
 
+hts-$(LIBHTS_SOVERSION).def: hts-$(LIBHTS_SOVERSION).dll
+	gendef hts-$(LIBHTS_SOVERSION).dll
+
+hts-$(LIBHTS_SOVERSION).lib: hts-$(LIBHTS_SOVERSION).def
+	llvm-dlltool -m i386:x86-64 -d hts-$(LIBHTS_SOVERSION).def -l hts-$(LIBHTS_SOVERSION).lib
+
+# Bundling libraries, binaries, dll dependencies, and licenses into a
+# single directory.
+#
+# NOTE: only tested on the supported MSYS2/MINGW64 environment.
+win-dist: all
+	$(MAKE) install prefix=win-dist
```

Comment:
I wonder if this could be done by depending on `install` as a prerequisite and documenting conventions around using DESTDIR instead. Calling `$(MAKE)` is a bit unfortunate and suggests this would go better as a _contrib/*_ script IMHO. Or if it stays as recursive make, the recipe line should have a `+` prefix.

Reply:
Valid point on the "+" prefix, but I don't see the issue with a recursive make.  We already do this elsewhere infact (destdir).

Note I'm not actually expecting this target to be used much by users.  As commented it's mainly as a tool to aid the maintainers in building binary distributions for testing against other compilers, as possibly as a precursor to making binary distributions.

Personally I don't like moving these sorts of things into external scripts either.  It's not how the GNU tools work either.   The general wisdom there is with "make dist" and "make distcheck" targets, done in-situ within the Makefile.   Given the simplicity of this and the limited audience for now I think it's appropriate as is.

Reply:
Ah I think I see your meaning about DESTDIR.  So something like:

```
dist-windows: DESTDIR=win-dist
dist-windows: install
        # stuff
```

It's possible a target-specific variable may work, but I'd need to read up on how portible it is.  Hmm actually, maybe not... our only supported windows environment is MSYS/MINGW which is probably guaranteed to be GNU make.

Reply:
I was imagining making you type `make DESTDIR=disttemp dist-windows` rather than the target-specific variable, but that (target-specific variable) is probably the best approach even though they're [a bit fragile](https://stackoverflow.com/questions/3261737/change-makefile-variable-value-depending-on-a-target/3263435#3263435).

Fair enough re in situ rather than a separate script. (That `$(MAKE)` in `distdir` is there because it's needed to cooperate with automake. Also Ed made me do it :smile:)